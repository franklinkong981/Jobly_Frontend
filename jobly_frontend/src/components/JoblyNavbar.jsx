import React, {useContext} from "react";
import { NavLink } from "react-router-dom";
import { Navbar, Nav, NavItem } from "reactstrap";

function JoblyNavbar() {
  const {currentUserInfo} = useContext(CountContext);

  return (
    <div>
      <Navbar expand="md">
        <NavLink exact to="/" className="navbar-brand">
          Jobly
        </NavLink>

        {currentUserInfo ? (
          <Nav className="ml-auto" navbar>
            <NavItem>
              <NavLink to="/companies">Companies</NavLink>
            </NavItem>
            <NavItem>
              <NavLink to="/jobs">Jobs</NavLink>
            </NavItem>
            <NavItem>
              <NavLink to="/profile">Update Profile</NavLink>
            </NavItem>
            <NavItem>
              <NavLink to="/logout">Log Out</NavLink>
            </NavItem>
          </Nav>
        ) : (
          <Nav className="ml-auto" navbar>
            <NavItem>
              <NavLink to="/login">Log In</NavLink>
            </NavItem>
            <NavItem>
              <NavLink to="/signup">Sign Up</NavLink>
            </NavItem>
          </Nav>
        )}
      </Navbar>
    </div>
  );
}

export default JoblyNavbar;