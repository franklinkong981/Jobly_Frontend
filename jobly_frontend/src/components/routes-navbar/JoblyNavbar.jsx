import React, {useContext} from "react";
import { Link, NavLink } from "react-router-dom";

import CurrentUserContext from "../../contexts/currentUserContext.jsx";

function JoblyNavbar({logOutFunc}) {
  const {currentUserInfo} = useContext(CurrentUserContext);

  function loggedInNavbar() {
    return (
      <ul className="navbar-nav ml-auto">
        <li className="nav-item mr-4">
          <NavLink className="nav-link" to="/companies">
            All Companies
          </NavLink>
        </li>
        <li className="nav-item mr-4">
          <NavLink className="nav-link" to="/jobs">
            All Jobs
          </NavLink>
        </li>
        <li className="nav-item mr-4">
          <NavLink className="nav-link" to="/profile">
            Update Profile
          </NavLink>
        </li>
        <li className="nav-item">
          <Link className="nav-link logout-link" to="/" onClick={logOutFunc}>
            Log out {currentUserInfo.username}
          </Link>
        </li>
      </ul>
    );
  }

  function loggedOutNavbar() {
    return (
      <ul className="navbar-nav ml-auto">
        <li className="nav-item mr-4">
          <NavLink className="nav-link" to="/login">
            Log In
          </NavLink>
        </li>
        <li className="nav-item mr-4">
          <NavLink className="nav-link" to="/signup">
            Sign Up
          </NavLink>
        </li>
      </ul>
    );
  }

  return (
    <nav className="JoblyNavbar navbar navbar-expand-md">
      <Link className="navbar-brand" to="/">
        Jobly
      </Link>
      {currentUserInfo ? loggedInNavbar() : loggedOutNavbar()}
    </nav>
  );
}

export default JoblyNavbar;