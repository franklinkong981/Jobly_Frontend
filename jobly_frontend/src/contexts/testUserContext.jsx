import React from "react";
import CurrentUserContext from "./currentUserContext.jsx";

const testUser = {
  username: "testuser",
  first_name: "Testy",
  last_name: "Usery",
  email: "testuser981@gmail.com"
};

const TestUserProvider = ({children, currentUserInfo = testUser, hasUserAppliedToJob= () => false}) => (
  <CurrentUserContext.Provider value={{currentUserInfo, hasUserAppliedToJob}}>
    {children}
  </CurrentUserContext.Provider>
);

export {TestUserProvider};