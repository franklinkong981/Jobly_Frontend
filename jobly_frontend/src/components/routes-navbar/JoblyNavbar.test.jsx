import React from "react";
import {render} from "@testing-library/react";
import {MemoryRouter} from "react-router-dom";

import JoblyNavbar from "./JoblyNavbar.jsx";

import {TestUserProvider} from "../../contexts/testUserContext.jsx";
import { expect, it } from "vitest";

it("renders without crashing", function() {
  render(
    <MemoryRouter>
      <TestUserProvider>
        <JoblyNavbar />
      </TestUserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot when logged in", function() {
  const {asFragment} = render(
    <MemoryRouter>
      <TestUserProvider>
        <JoblyNavbar />
      </TestUserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function() {
  const {asFragment} = render(
    <MemoryRouter>
      <TestUserProvider currentTestUser={null}>
        <JoblyNavbar />
      </TestUserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});