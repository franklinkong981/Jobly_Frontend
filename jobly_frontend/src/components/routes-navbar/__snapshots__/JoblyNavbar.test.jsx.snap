// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot when logged in 1`] = `
<DocumentFragment>
  <nav
    class="JoblyNavbar navbar navbar-expand-md"
  >
    <a
      class="navbar-brand"
      data-discover="true"
      href="/"
    >
      Jobly
    </a>
    <ul
      class="navbar-nav ml-auto"
    >
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          data-discover="true"
          href="/companies"
        >
          All Companies
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          data-discover="true"
          href="/jobs"
        >
          All Jobs
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          data-discover="true"
          href="/profile"
        >
          Update Profile
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link logout-link"
          data-discover="true"
          href="/"
        >
          Log out testuser
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <nav
    class="JoblyNavbar navbar navbar-expand-md"
  >
    <a
      class="navbar-brand"
      data-discover="true"
      href="/"
    >
      Jobly
    </a>
    <ul
      class="navbar-nav ml-auto"
    >
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          data-discover="true"
          href="/companies"
        >
          All Companies
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          data-discover="true"
          href="/jobs"
        >
          All Jobs
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          data-discover="true"
          href="/profile"
        >
          Update Profile
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link logout-link"
          data-discover="true"
          href="/"
        >
          Log out testuser
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
